pandas/tests/plotting/frame/test_hist_box_by.py

```diff
@@ -0,0 +1,289 @@
+import re
+
+import numpy as np
+import pytest
+
+import pandas.util._test_decorators as td
+
+from pandas import DataFrame
+import pandas._testing as tm
+from pandas.tests.plotting.common import (
+    TestPlotBase,
+    _check_plot_works,
+)
+
+
+def test_hist_box_with_by_df():
+    np.random.seed(0)
+    df = DataFrame(np.random.randn(30, 2), columns=["A", "B"])
+    df["C"] = np.random.choice(["a", "b", "c"], 30)
+    df["D"] = np.random.choice(["a", "b", "c"], 30)
+    return df
```

Comment:
If you start this with `test_` pytest will execute it, and it doesn't feel that's what you want, is it?

Reply:
Ahh, nice catch! I renamed it! Thanks!

Reply:
No big deal, but if this function is not used outside of the file, and not expected to be used by users (since this is a tests file), better to start the function name with underscore. This way other devs know this is not used anywhere else.